packages/core/core/src/AssetGraph.js

```diff
@@ -265,7 +265,13 @@ export default class AssetGraph extends Graph<AssetGraphNode> {
 
     let res: ?Asset = null;
     this.traverse((node, ctx, traversal) => {
-      if (node.type === 'asset' || node.type === 'asset_reference') {
+      // Prefer real assets when resolving dependencies, but use the first
```

Comment:
This allows the runtime to inject the js asset which exports the url, and have it properly resolve in the packager. Otherwise, the asset reference to the "real" file would be resolved.

An alternative would be to reorder the edges so that the one pointing to the js asset would come first, but that seems clumsy.